lib/experiment/local/client.rb

```diff
@@ -0,0 +1,79 @@
+require 'uri'
+require 'logger'
+require 'experiment/local/evaluation/evaluation'
+
+module AmplitudeExperiment
+  # Main client for fetching variant data.
+  class LocalEvaluationClient
+    # Creates a new Experiment Client instance.
+    #
+    # @param [String] api_key The environment API Key
+    # @param [LocalEvaluationConfig] config The config object
+    def initialize(api_key, config = nil)
+
+      @api_key = api_key
+      @config = config || LocalEvaluationConfig.new
+      @cache = InMemoryFlagConfigCache.new(@config.bootstrap)
+      @logger = Logger.new($stdout)
+      @logger.level = if @config.debug
+                        Logger::DEBUG
+                      else
+                        Logger::INFO
+                      end
+      endpoint = "#{@config.server_url}/sdk/vardata"
+      @uri = URI(endpoint)
+      @fetcher = LocalEvaluationFetcher.new(api_key, @config.debug)
+      @poller = FlagConfigPoller.new(@fetcher, @cache, @config.debug)
+
+      raise ArgumentError, 'Experiment API key is empty' if @api_key.nil? || @api_key.empty?
+    end
+
+    # Locally evaluates flag variants for a user.
+    #
+    # @param [User] user The user to evaluate
+    # @param [String[]] flag_keys The flags to evaluate with the user. If empty, all flags from the flag cache are evaluated
+    #
+    # @return [Variants] The evaluated variants
+    def evaluate(user = User.new, flag_keys = [])
+      flag_configs = []
+      if flag_keys.empty?
+        @cache.get_all.each do |key, value|
+          flag_configs.push(value)
+        end
+      else
+        flag_configs = get_flag_configs(flag_keys)
+      end
+      flag_configs_str = flag_configs.to_json.inspect.gsub(/\\/, '')[1..].chop!
```

Comment:
to_json will convert the object to string with `\` spend before " and the double quote " at the outer level. It will throw error in evaluation logic. So I did the trick here to remove `\`, the leading " and ending ".
Should we make improvements in the chore evaluation to handle this? Or is there any other way to do it in Ruby?

example:
[{"allUsersTargetingConfig":{"allocations":xxx}].to_json == "[{\"allUsersTargetingConfig\":{\"allocations\":xxx}]"
Evaluation pass through when change the string to [{'allUsersTargetingConfig':{'allocations':xxx}]